File: typescript/base-hmi-types/src/screens/base/HmiThickness.ts
import { HmiProperty, staticProperty } from "./HmiProperty.js";

export class HmiThickness {
  left: HmiProperty<number> = staticProperty(0);
  top: HmiProperty<number> = staticProperty(0);
  right: HmiProperty<number> = staticProperty(0);
  bottom: HmiProperty<number> = staticProperty(0);
}

File: src/BaseHmiTypes/Screens/Base/HmiThickness.cs
namespace BaseHmiTypes.Screens.Base;

public class HmiThickness
{
    public HmiProperty<double> Left { get; set; } = 0;

    public HmiProperty<double> Top { get; set; } = 0;

    public HmiProperty<double> Right { get; set; } = 0;

    public HmiProperty<double> Bottom { get; set; } = 0;
}
